src/Commands/utilities/commands.js

```diff
@@ -122,36 +128,28 @@ Permissions: ${c.permission}
 }
 
 // this function gets the slash command, now i only need to make it get if the command has sub commands
-// TY Sister https://discord.com/channels/1055188344188973066/1096722788385050694/1199455671016509480
-async function getSlashCommand(client, name, subcommand) {
-
-	if (!client.application.commands.cache.size)
-		await client.application.commands.fetch();
-
+// TY Blåhaj https://discord.com/channels/1055188344188973066/1096722788385050694/1199455671016509480
+function getSlashCommand(client, name, subcommand) {
 	const command = client.application.commands.cache.find(((cmd) => cmd.name === name));
 	if (!command) return `\`${name}${subcommand ? ` ${subcommand}` : ""}\``;
 
 	return `</${command.name}${subcommand ? ` ${subcommand}` : ""}:${command.id}>`
 }
 
-async function getSubcommand(interaction, list, name) {
-
-	if (!interaction.client.application.commands.cache.size)
-		await interaction.client.application.commands.fetch();
-
-	for await (const [one, command] of interaction.client.application.commands.cache) {
-		if (command.name === name) {
-
-			const subcommands = (command?.options)
-				? (command?.options || []).filter(option => option.type === ApplicationCommandOptionType.Subcommand)
-				: [];
-			if (subcommands.length) {
-				for await (const subcommand of subcommands) {
-					list.push(`${await getSlashCommand(interaction.client, command.name, subcommand.name)}`);
-				}
-				continue;
+function getSubcommand(interaction, list, name) {
+	for (const [, command] of interaction.client.application.commands.cache) {
+		if (command.name !== name) continue;
+		
+		const subcommands = (command?.options || []).filter(option => option.type === ApplicationCommandOptionType.Subcommand) || [];
+		
+		if (subcommands.length) {
+			for (const subcommand of subcommands) {
+				list.push(`${getSlashCommand(interaction.client, command.name, subcommand.name)}`);
 			}
-			list.push(await getSlashCommand(interaction.client, command.name));
 		}
+		
+		list.push(getSlashCommand(interaction.client, command.name));
 	}
-}
\ No newline at end of file
+
+	return list;

```

Comment:
You didn't return before but I assume you meant to do so, else this function didn't even do anything usefull.